src/plugins/home/server/services/new_instance_status.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IScopedClusterClient, SavedObjectsClientContract } from '../../../../core/server';
+import type { IndexPatternSavedObjectAttrs } from '../../../data/common/index_patterns/index_patterns';
+
+const LOGS_INDEX_PATTERN = 'metrics-*';
+const METRICS_INDEX_PATTERN = 'logs-*';
+
+const INDEX_PREFIXES_TO_IGNORE = [
+  'metrics-endpoint.metadata-', // ignore index created by default by endpoint transform
+  'metrics-elastic_agent', // ignore index created by Fleet server itself
+  'logs-elastic_agent', // ignore index created by Fleet server itself
+];
+
+interface Deps {
+  esClient: IScopedClusterClient;
+  soClient: SavedObjectsClientContract;
+}
+
+export const isNewInstance = async ({ esClient, soClient }: Deps): Promise<boolean> => {
+  const indexPatterns = await soClient.find<IndexPatternSavedObjectAttrs>({
+    type: 'index-pattern',
+    fields: ['title'],
+    search: `*`,
+    searchFields: ['title'],
+    perPage: 1,
+  });
+
+  // If there are no index patterns, assume this is a new instance
+  if (indexPatterns.total === 0) {
+    return true;
+  }
+
+  // If there are any index patterns that are not the default metrics-* and logs-* ones created by Fleet, assume this
+  // is not a new instance
+  if (
+    indexPatterns.saved_objects.some(
+      (ip) =>
+        ip.attributes.title !== LOGS_INDEX_PATTERN && ip.attributes.title !== METRICS_INDEX_PATTERN
+    )
+  ) {
+    return false;
+  }
+
+  try {
+    const logsAndMetricsIndices = await esClient.asCurrentUser.cat.indices({
```

Comment:
would this work in CCS environments?

Reply:
Turns out that the `/_cat/indices` API does not support CCS. We have a few options here:
1. Don't handle this case. I don't think it's likely that a user will have configured remote clusters for CCS before they've ever opened Kibana. I also think that if they have set up CCS, they still need to configure the index patterns for CCS (they're not setup for CCS by default). With the current logic, they'll be prompted to add data in this case. This does mean that in the case where a new space is created that doesn't have index patterns on a cluster with CCS, we'll show them this modal. This was also the behavior before this broke in 7.13.
2. Use the `/_search` API instead of the `/_cat/indices` API. This is going to make this check slower, especially in the case where there are remote clusters configured. This could be confusing since it would delay the display of this UI, either leaving the user with a blank loading UI or displaying the Home page briefly and then covering it up with this UI.

With the priority of this change being on supporting the default onboarding experience for new users, I don't think handling CCS is necessary.